### PR TITLE
Adds configurable stop tokens

### PR DESCRIPTION
**Pull request by zmackie**
This is an initial start of #224 
I wanted to open this PR to make sure the maintainers were okay with the direction.
`stop` is sort of an odd parameter that doesn't have an obvious place to live (from what I saw...very well could be wrong).

Thanks!

### CODE REVIEW

**Comment by zmackie** [comment]
Oh and my apologies for missing the DCO. Will address that tomorrow.

Zander Mackie

File: nemoguardrails/llm/taskmanager.py [hunk 1]
```diff
@@ -255,6 +255,11 @@ def parse_task_output(self, task: Task, output: str):
         else:
             return output
 
+    def task_stop(self, task: Task) -> List[str]:
```
Comment (drazvan) [inline nemoguardrails/llm/taskmanager.py]:
```suggestion
    def task_stop(self, task: Union[str, Task]) -> List[str]:
```
This should work with both `str` and `Task` in the same way `render_task_prompt` does.

Comment (drazvan) [reply nemoguardrails/llm/taskmanager.py]: And maybe rename to `get_stop_tokens` for better readability.

**Review by drazvan — COMMENTED** [review]
Looks good 👍 . This is the right direction.

**Comment by zmackie** [comment]
@drazvan I believe this should do it. I wasn't totally sure about the use of stops in generation tasks, but I guessed this wasn't useful. LMK what you think. Happy to update documentation, etc. if that's important.

**Comment by drazvan** [comment]
Let's merge this. I'll add the documentation in a separate commit.

**Pull request merged**